src/test/java/org/gridsuite/shortcircuit/server/service/ShortCircuitServiceTest.java

```diff
@@ -1,179 +1,338 @@
-import org.gridsuite.shortcircuit.server.computation.service.ExecutionService;
 import org.gridsuite.shortcircuit.server.computation.service.NotificationService;
-import org.gridsuite.shortcircuit.server.computation.service.ReportService;
-import org.gridsuite.shortcircuit.server.reports.AbstractReportMapper;
-import org.junit.jupiter.api.BeforeEach;
+import org.gridsuite.shortcircuit.server.computation.service.UuidGeneratorService;
+import org.gridsuite.shortcircuit.server.dto.ShortCircuitParametersInfos;
+import org.gridsuite.shortcircuit.server.dto.ShortCircuitPredefinedConfiguration;
+import org.gridsuite.shortcircuit.server.entities.AnalysisParametersEntity;
+import org.gridsuite.shortcircuit.server.repositories.ParametersRepository;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
 import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.Mock;
-import org.mockito.MockedStatic;
+import org.mockito.ArgumentCaptor;
 import org.mockito.Mockito;
 import org.mockito.junit.jupiter.MockitoExtension;
-import org.springframework.messaging.Message;
-import org.springframework.messaging.support.GenericMessage;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
 
-import java.util.Collections;
-import java.util.List;
+import java.util.Optional;
+import java.util.Random;
 import java.util.UUID;
-import java.util.concurrent.CompletableFuture;
-import java.util.stream.Stream;
 
-import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.*;
 
 @ExtendWith({ MockitoExtension.class })
-@Slf4j
+@MockitoSettings(strictness = Strictness.STRICT_STUBS)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 class ShortCircuitServiceTest implements WithAssertions {
-
-    @Mock
-    private NetworkStoreService networkStoreService;
-
-    @Mock
-    private ReportService reportService;
-
-    @Mock
-    private ExecutionService shortCircuitExecutionService;
-
-    @Mock
     private NotificationService notificationService;
-
-    @Mock
-    private ShortCircuitAnalysisResultService resultRepository;
-
-    @Mock
+    private UuidGeneratorService uuidGeneratorService;
+    private ShortCircuitAnalysisResultService resultService;
+    private ParametersRepository parametersRepository;
     private ObjectMapper objectMapper;
+    private ShortCircuitService shortCircuitService;
 
-    @Mock
-    private AbstractReportMapper reportMapper;
+    @BeforeAll
+    void setUp() {
+        this.notificationService = mock(NotificationService.class);
+        this.uuidGeneratorService = spy(new UuidGeneratorService());
+        this.resultService = mock(ShortCircuitAnalysisResultService.class);
+        this.parametersRepository = mock(ParametersRepository.class);
+        this.objectMapper = spy(new ObjectMapper());
```

Comment:
Why not annotations ?

Reply:
* `@springBootTest`: not a Spring test, I don't need it
* `@ExtendWith(MockitoExtension. class)`: I tried but `@Mock`s and `@Spy`s are create and destroy in (Before/After)Each phase, so I have null objects in BeforeAll step.